apps/query-ui/queryui/util.py

```diff
@@ -85,86 +143,136 @@ def docset_to_string(docset: DocSet) -> str:
     return retval
 
 
-def show_dag(plan: LogicalPlan):
-    nodes = []
-    edges = []
-    for node in plan.nodes.values():
-        assert isinstance(node, LogicalOperator)
-        nodes.append(
-            Node(
-                id=node.node_id,
-                label=f"[Node {node.node_id}] {type(node).__name__}\n\n{node.description}",
-                shape="box",
-                color={
-                    "background": "#404040",
-                    "border": "#5050f0",
-                },
-                font="14px arial white",
-                chosen=False,
-                margin=30,
-            )
-        )
-    for node in plan.nodes.values():
-        if node.dependencies:
-            for dep in node.dependencies:
-                edges.append(Edge(source=dep.node_id, target=node.node_id, color="#ffffff"))
-
-    config = Config(
-        width=700,
-        height=500,
-        directed=True,
-        physics=False,
-        hierarchical=True,
-        direction="UD",
-    )
-    agraph(nodes=nodes, edges=edges, config=config)
-
-
-@st.experimental_fragment
-def show_query_traces(trace_dir: str, query_id: str):
-    """Show the query traces in the given trace_dir."""
-    trace_dir = os.path.join(trace_dir, query_id)
-    for node_id in sorted(os.listdir(trace_dir)):
-        data_list = []
-        directory = os.path.join(trace_dir, node_id)
+class QueryNodeTrace:
+    """Display query traces for a single query node."""
+
+    # The order here is chosen to ensure that the most important columns are shown first.
+    # The logic below ensure that additional columns are also shown, and that empty columns
+    # are omitted.
+    COLUMNS = [
+        "properties.path",
+        "properties.entity.accidentNumber",
+        "properties.key",
+        "properties.count",
+        "properties.entity.location",
+        "properties.entity.dateAndTime",
+        "properties.entity.aircraft",
+        "properties.entity.registration",
+        "properties.entity.conditions",
+        "properties.entity.windSpeed",
+        "properties.entity.visibility",
+        "properties.entity.lowestCeiling",
+        "properties.entity.injuries",
+        "properties.entity.aircraftDamage",
+        "text_representation",
+        "doc_id",
+        "parent_id",
+    ]
+
+    def __init__(self, trace_dir: str, node_id: str):
+        self.trace_dir = trace_dir
+        self.node_id = node_id
+        self.df = None
+        self.readdata()
```

Comment:
I like calling internal methods with an _ to make it clear they aren't part of the public API. Here I think thats _readdata() and _readfile().